python/lsst/afw/image/image.cc

```diff
@@ -21,25 +21,158 @@
+            "fileName"_a, "metadata"_a=CONST_PTR(lsst::daf::base::PropertySet)(), "mode"_a="w");
+    cls.def("writeFits", (void (Image<PixelT>::*)(lsst::afw::fits::MemFileManager &, CONST_PTR(lsst::daf::base::PropertySet), std::string const&) const) &Image<PixelT>::writeFits,
+            "manager"_a, "metadata"_a=CONST_PTR(lsst::daf::base::PropertySet)(), "mode"_a="w");
+    cls.def("writeFits", (void (Image<PixelT>::*)(lsst::afw::fits::Fits &, CONST_PTR(lsst::daf::base::PropertySet)) const) &Image<PixelT>::writeFits,
+            "fitsfile"_a, "metadata"_a=CONST_PTR(lsst::daf::base::PropertySet)());
+    cls.def_static("readFits", (Image<PixelT> (*)(std::string const &, int)) Image<PixelT>::readFits,
+            "filename"_a, "hdu"_a=0);
+    cls.def_static("readFits", (Image<PixelT> (*)(lsst::afw::fits::MemFileManager &, int)) Image<PixelT>::readFits,
+            "manager"_a, "hdu"_a=0);
+    cls.def("sqrt", &Image<PixelT>::sqrt);
+
+    /* Add-ons for Python interface only */
+    cls.def("set", [](Image<PixelT> &img, double val) { img=val; });
+    cls.def("set", [](Image<PixelT> &img, int x, int y, double val) { img(x, y, lsst::afw::image::CheckIndices(true))=val; });
+    cls.def("get", [](Image<PixelT> &img, int x, int y) { return img(x, y, lsst::afw::image::CheckIndices(true)); });
+    cls.def("set0", [](Image<PixelT> &img, int x, int y, double val) { img.set0(x, y, val, lsst::afw::image::CheckIndices(true)); });
+    cls.def("set0", [](Image<PixelT> &img, int x, int y) { return img.get0(x, y, lsst::afw::image::CheckIndices(true)); });
+
+    return cls;
+}
+
+/* Declare ImageSlice operators separately since they are only instantiated for Image<float> and Image<double> */
+template <typename PixelT>
+void addImageSliceOperators(py::class_<Image<PixelT>> & cls) {
+    cls.def("__add__", [](Image<PixelT> const & self, ImageSlice<PixelT> const & other) { return self + other; }, py::is_operator());
+    cls.def("__sub__", [](Image<PixelT> const & self, ImageSlice<PixelT> const & other) { return self - other; }, py::is_operator());
+    cls.def("__mul__", [](Image<PixelT> const & self, ImageSlice<PixelT> const & other) { return self * other; }, py::is_operator());
+    cls.def("__truediv__", [](Image<PixelT> const & self, ImageSlice<PixelT> const & other) { return self / other; }, py::is_operator());
+    cls.def("__iadd__", [](Image<PixelT> & self, ImageSlice<PixelT> const & other) { self += other; return self; }, py::is_operator());
+    cls.def("__isub__", [](Image<PixelT> & self, ImageSlice<PixelT> const & other) { self -= other; return self; }, py::is_operator());
+    cls.def("__imul__", [](Image<PixelT> & self, ImageSlice<PixelT> const & other) { self *= other; return self; }, py::is_operator());
+    cls.def("__itruediv__", [](Image<PixelT> & self, ImageSlice<PixelT> const & other) { self /= other; return self; }, py::is_operator());
+}
```

Comment:
Hmm, there better be a nice way to do this because otherwise it is going to be a mess.